src/main.rs

```diff
@@ -380,11 +434,90 @@ fn main() {
 					},
 				}
 			}
-			println!("Building for {}...", MinSec::since(starttime));
-			if workers.iter().all(|worker| *worker == WorkerStatus::Done) {
+
+			let build_is_done = buildstate.all_workers_done();
+
+			// Compute remaining time for this build, by simulating a build.
+			let mut simulated_time = Instant::now();
+			let mut estimation_impossible = false;
+			while !buildstate.all_workers_done_or_idle() {
+				// Give all simulated workers something to do
+				for i in 0..buildstate.workers.len() {
+					let worker = &buildstate.workers[i];
+					match worker {
+						WorkerStatus::Starting | WorkerStatus::Idle => {
+							let next = queuestate.next_at(simulated_time);
+							buildstate.set_status(i, match next {
+								Some(task) => WorkerStatus::Running{task},
+								None => WorkerStatus::Idle,
+							});
+						}
+						_ => {}
+					}
+				}
+
+				// Find the job with the lowest remaining time
+				let earliest_remaining_job = match buildstate.workers.iter()
+					.enumerate()
+					.flat_map(|(i, worker)| {
+						match worker {
+							WorkerStatus::Running{task, ..} => {
+								let taskstatus = queuestate.get_task_status(*task);
+								match taskstatus {
+									TaskStatus::Running{start_time, ..} => {
+										let runtime = if simulated_time > start_time { simulated_time - start_time } else { Duration::from_millis(0) };
```

Comment:
You can use [`checked_sub`](https://doc.rust-lang.org/std/time/struct.Duration.html#method.checked_sub) here: `simulated_time.checked_sub(start_time).unwrap_or(Duration::from_millis(0))`. (Unfortunately, `Duration` does not have a `saturating_sub` (yet) like the integer types.)